pkg/cmd/admin/router/router.go

```diff
@@ -457,10 +464,17 @@ func RunCmdRouter(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg *
 		return fmt.Errorf("unable to parse --ports: %v", err)
 	}
 
-	// For the host networking case, ensure the ports match. Otherwise, remove host ports
-	for i := 0; i < len(ports); i++ {
-		if cfg.HostNetwork && ports[i].HostPort != 0 && ports[i].ContainerPort != ports[i].HostPort {
-			return fmt.Errorf("when using host networking mode, container port %d and host port %d must be equal", ports[i].ContainerPort, ports[i].HostPort)
+	// HostNetwork overrides HostPorts
+	if cfg.HostNetwork {
+		cfg.HostPorts = false
```

Comment:
unfortunate we can't tell here if the user explicitly passed `--host-network --host-ports` (which should probably be a validation error?) or if we're just adjusting the default